### PR TITLE
announce 4.52

### PR DESCRIPTION
**Pull request by fredemmott**

### CODE REVIEW

File: _posts/2020-04-07-hhvm-4.52.markdown [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# Highlights
+
+- fixed crash in `ext_soap` if a SoapServer's handler class does not exist.
+- `?>` and `<?=` are no longer recognized by the parser; this has replaced
+  dedicated errors with parse errors.
+
+# Breaking Changes
+
+- `__halt_compiler()` has been removed.
+- it is now a type error to `use` the same trait twice in the same class.
```
Comment (lexidor) [inline _posts/2020-04-07-hhvm-4.52.markdown:20]: Does this include via inheritance?
E.g. `BaseClass` uses `SomeTrait` and `ChildClass` uses `SomeTrait` and `ChildClass` extends `BaseClass`.

**Comment by fredemmott** [comment]
Holding off on publishing: the typechecker is now raising parse errors for actual PHP files, eg vendor/autoload.php. Need that fixed.

File: _posts/2020-04-07-hhvm-4.52.markdown [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Highlights
+
+- fixed crash in `ext_soap` if a SoapServer's handler class does not exist.
+- `?>` and `<?=` are no longer recognized by the parser; this has replaced
+  dedicated errors with parse errors.
+
+# Breaking Changes
+
+- `__halt_compiler()` has been removed.
```
Comment (jjergus) [inline _posts/2020-04-07-hhvm-4.52.markdown:19]: I think it's been removed months ago (I see it announced in https://hhvm.com/blog/2019/12/10/hhvm-4.35.html), the current change might just be that it's no longer recognized by the parser so no longer gives a dedicated error (will just be "undefined function" now)?

**Comment by fredemmott** [comment]
Kicked off a build of 4.52.1, with a fix for `hh_client` raising parse errors for real php files such as vendor/autoload.php

File: _posts/2020-04-07-hhvm-4.52.markdown (hunk 1 above)
Comment (fredemmott) [reply _posts/2020-04-07-hhvm-4.52.markdown:20]: nope, will expand

Comment (Wilfred) [reply _posts/2020-04-07-hhvm-4.52.markdown]: It does not, but although there is a typechecker option `no_trait_reuse` to forbid this. That might become the default at some point.

File: _posts/2020-04-07-hhvm-4.52.markdown (hunk 2 above)
Comment (Wilfred) [reply _posts/2020-04-07-hhvm-4.52.markdown]: Yep, that's correct.

**Pull request merged**